src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -6792,8 +7337,35 @@ instruct addL_rReg_imm(rRegL dst, immL32 src, rFlagsReg cr)
   ins_pipe( ialu_reg );
 %}
 
+instruct addL_rReg_rReg_imm_ndd(rRegL dst, rRegL src1, immL32 src2, rFlagsReg cr)
+%{
+  predicate(UseAPX);
+  match(Set dst (AddL src1 src2));
+  effect(KILL cr);
+
+  format %{ "eaddq    $dst, $src1, $src2\t# long ndd" %}
+  ins_encode %{
+    __ eaddq($dst$$Register, $src1$$Register, $src2$$constant, false);
+  %}
+  ins_pipe( ialu_reg );
+%}
+
+instruct addL_rReg_mem_imm_ndd(rRegL dst, memory src1, immL32 src2, rFlagsReg cr)
+%{
+  predicate(UseAPX);
+  match(Set dst (AddL (LoadL src1) src2));
+  effect(KILL cr);
+
+  format %{ "eaddq    $dst, $src1, $src2\t# long ndd" %}
```

Comment:
Can you kindly share the impact of adding all these new patterns on libjvm.so size ?

I am curious to know the amount of ADLC generated code for all these new patterns.